scripts/build/utils.js

```diff
@@ -38,12 +38,27 @@ export function getPackageJson(rootPath, prod = false) {
   const cjsDir = getCJSDir();
   const esmDir = getESMDir();
 
-  /** @param {string} path */
-  const getExports = (path) => {
+  /**
+   * @param {string} path
+   * @param {boolean} includeTypes
+   */
+  const getExports = (path, includeTypes) => {
     if (!prod) {
       return path.replace(sourcePath, `./${sourceDir}`);
     }
     path = removeExt(path).replace(sourcePath, "");
+    if (includeTypes) {
+      return {
+        import: {
+          default: `./${join(esmDir, path)}.js`,
+          types: `./${join(cjsDir, path)}.d.ts`,
+        },
+        require: {
+          default: `./${join(cjsDir, path)}.cjs`,
+          types: `./${join(cjsDir, path)}.d.cts`,
+        },
+      };
+    }
```

Comment:
According to the [TypeScript documentation](https://www.typescriptlang.org/docs/handbook/esm-node.html#packagejson-exports-imports-and-self-referencing), the `types` field should be listed first. However, the documentation also notes that it will search for a declaration file in the same location. So, I'm curious whether it's necessary to explicitly set it here.

Reply:
It is. But I could certainly flip the order. In terms of repro, should I do that just as a separate git repo or is there a more typical place to do that in ariakit? i.e. do we use stackblitz or some similar thing.

Reply:
> It is.

Is it still necessary if you remove the `.d.ts` files from the folder, only leaving the `.d.cts` ones?

Reply:
If we only had the .d.cts, then we wouldn't be pointing to in properly in the global "types" key. All of this is premised on the idea that upstream users of ariakit should not be forced into type: module, as evidenced by the multiple build outputs and the multiple import paths. This PR just fixes that to include Typescript situations that do not rely on ESM.